.chloggen/redaction-blocked-patterns.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: processor/redaction
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Introduce 'blocked_key_patterns' and support hashing instead of masking values via 'hash_function' parameter"
```

Comment:
This feels like two independent issues. Could we split this PR into two?

Reply:
adapted, please have another look and keep in mind that this PR needs to be merged before https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38161 due to the refactoring the the tests made